### PR TITLE
fix(aws-s3-cloudfront): Recognize when client specifies enforceSSL: false

### LINKED ISSUE #518: CloudFrontToS3 does not produce correct bucket policy when "enforceSSL" is false
Using `@aws-solutions-constructs/aws-cloudfront-s3@1.121.0` and setting `bucketProps.enforceSSL` to `false` and `bucketProps.publicReadAccess` to `true` (for a public facing website), I get a bucket policy that produces a 403 when I attempt to access the website:

```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Principal": {
                "AWS": "*"
            },
            "Action": "s3:GetObject",
            "Resource": "arn:aws:s3::BUCKET_NAME/*"
        },
        {
            "Sid": "HttpsOnly",
            "Effect": "Deny",
            "Principal": {
                "AWS": "*"
            },
            "Action": "*",
            "Resource": [
                "arn:aws:s3:::BUCKET_NAME/*",
                "arn:aws:s3:::BUCKET_NAME"
            ],
            "Condition": {
                "Bool": {
                    "aws:SecureTransport": "false"
                }
            }
        }
    ]
}
```

The correct bucket policy should have the `HttpsOnly` Statement omitted.

### Reproduction Steps

```typescript
    const webui = new CloudFrontToS3(this, "WebUI", {
      bucketProps: {
        publicReadAccess: true,
        enforceSSL: false,
        websiteErrorDocument: "index.html",
        websiteIndexDocument: "index.html",
        serverAccessLogsPrefix: "access-logs",
        lifecycleRules: [
          {
            enabled: true,
            expiration: Duration.days(30),
            prefix: "access-logs",
          },
        ],
        blockPublicAccess: new BlockPublicAccess({
          blockPublicAcls: false,
          blockPublicPolicy: false,
          ignorePublicAcls: false,
          restrictPublicBuckets: false,
        }),
        versioned: false,
      },
      cloudFrontDistributionProps: {
        certificate: certificate,
        defaultRootObject: "index.html",
        domainNames: ['website.domain.tld'],
        priceClass: PriceClass.PRICE_CLASS_100,
      } as DistributionProps,
      insertHttpSecurityHeaders: false, // Disable for now because the CSP breaks Angular
    })

    new BucketDeployment(this, "DeployWebUI", {
      sources: [Source.asset(path.resolve(__dirname, "../../frontend/dist/website"))],
      destinationBucket: webui.s3BucketInterface,
    })

    new ARecord(this, "frontendDNS", {
      zone,
      recordName: this.config.frontend.hostName,
      target: RecordTarget.fromAlias(new CloudFrontTarget(webui.cloudFrontWebDistribution)),
    })
```

### Error Log

403 Forbidden

*  Code: AccessDenied
*  Message: Access Denied
*  RequestId: *redacted*
*  HostId: *redacted*

An Error Occurred While Attempting to Retrieve a Custom Error Document

    Code: AccessDenied
    Message: Access Denied

### Environment

  - **CDK CLI Version      : 1.121.0**
  - **CDK Framework Version: 1.121.0**
  - **AWS Solutions Constructs Version     : 1.121.0**
  - **OS                   : MacOS Big Sur**
  - **Language             : TypeScript**

### Other

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by mickychetta**
*Issue #518 , if available:*

*Description of changes:*
-Added condition to omit HTTPS only requests when enforceSSL is false
-Created tests to check BucketPolicy is correct for enforceSSL prop

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

fixes #518

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: b9419115751bad36a6951a1386e20721a4e40438
* Result: SUCCEEDED
* [Build Logs](https://j8dgcooz5f.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=64ad8025-eac6-4887-b14a-e68eea9dd314%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v1
* Commit ID: b9419115751bad36a6951a1386e20721a4e40438
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=bba52a02-4636-40e8-92e4-ab6455ee9b9a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/core/test/s3-bucket.test.ts [hunk 1]
```diff
@@ -154,3 +154,177 @@ test('test createAlbLoggingBucket()', () => {
     BucketName: 'test-name'
   });
 });
+
+test('Test bucket policy that only accepts SSL requests only', () => {
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/core/test/s3-bucket.test.ts:158]: Let's add 2 more tests - 
-SSL is enforced when bucketProps is not provided
-SSL is enforced when bucketProps is provided, but enforceSSL is not set

When we're talking a security issue we want to ensure that every possibility is explicitly tested.

**Review by biffgaut — CHANGES_REQUESTED** [review]
Just a couple more tests

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 47ac4b38ae59b9953580cd49eedbd2d32b91bd14
* Result: SUCCEEDED
* [Build Logs](https://j8dgcooz5f.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4fee7bee-228e-4752-a14e-9f09b6203012%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v1
* Commit ID: 47ac4b38ae59b9953580cd49eedbd2d32b91bd14
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=2873c7f6-0753-49cc-9a09-a2a5feecf741%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/core/lib/s3-bucket-helper.ts [hunk 2]
```diff
@@ -166,8 +166,10 @@ export function buildS3Bucket(scope: Construct,
 
   customBucketProps = props.bucketProps ? overrideProps(customBucketProps, props.bucketProps) : customBucketProps;
 
-  const s3Bucket: s3.Bucket = new s3.Bucket(scope, _bucketId, customBucketProps);
+  // Set enforceSSL to always false to prevent duplicate policy statement
+  const s3Bucket: s3.Bucket = new s3.Bucket(scope, _bucketId, { ...customBucketProps, enforceSSL: false });
 
+  // Apple secure bucket policy through appleSecureBucketPolicy function and not with enforceSSL prop
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/core/lib/s3-bucket-helper.ts:172]: Pretty sure you meant "Apply"?

File: source/patterns/@aws-solutions-constructs/core/lib/s3-bucket-helper.ts [hunk 3]
```diff
@@ -166,8 +166,10 @@ export function buildS3Bucket(scope: Construct,
 
   customBucketProps = props.bucketProps ? overrideProps(customBucketProps, props.bucketProps) : customBucketProps;
 
-  const s3Bucket: s3.Bucket = new s3.Bucket(scope, _bucketId, customBucketProps);
+  // Set enforceSSL to always false to prevent duplicate policy statement
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/core/lib/s3-bucket-helper.ts:169]: I don't think we need ApplySecurePolicy anymore - we just need to make sure enforceSSL is always true unless the client explicitly says it should be false.

**Review by biffgaut — CHANGES_REQUESTED** [review]

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v1
* Commit ID: 5c5e12804379d202e7a47ee3261b6e325665c729
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=91d3369b-958b-478c-858c-0b339e25d07e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 5c5e12804379d202e7a47ee3261b6e325665c729
* Result: FAILED
* [Build Logs](https://j8dgcooz5f.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=6e7939cc-b343-4de9-a610-74023c4c57b7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 57eb8d182053477fdd1343896496c083f75ba68b
* Result: SUCCEEDED
* [Build Logs](https://j8dgcooz5f.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4deac898-d035-4711-b49b-3e4556804c3f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v1
* Commit ID: 57eb8d182053477fdd1343896496c083f75ba68b
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b020b133-519f-4535-a65f-81e4604b7fa6%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by biffgaut — APPROVED** [review]

**Pull request merged**